### DISCUSSION TITLE
Q: how Filled Size in Simulation is determined

### DISCUSSION THREAD

**Discussion by pakornavakul** [nkaz001/hftbacktest#86] [Q&A] [answered]
Hi Again, 
thanks for making project. This saves me monthsss of time, and thank you for your replies

I am trying to understand how partial filled size or exec_qty is determined. I read from [https://hftbacktest.readthedocs.io/en/latest/order_fill.html](url) and tried to go through your codes in partialfillexchange.py. I see that is it using min(q_qty, qty, order.leaves_qty), but when I put in ridiculous large order size, I still get fully filled. Perhaps my input data is incorrect? What do you think is a good way to debug this?

**Comment by pakornavakul** [comment, marked as answer]
I understand why it fully filled now. Current version does not count the trades size when the prices cross our orders price, same with market moves.

**Reply by nkaz001** [reply]
You're correct, since market-data replay-based backtesting cannot alter the market, if the market's best bid or ask crosses your order or the market trade completely crosses your order, the order cannot be left unfilled or partially filled. In my opinion, if you have to consider the partial fill, it may be a sign of market impact. I am still unsure, but you need to handle the situation carefully, compare the backtesting results with the actual live trading results, and tune the backtesting to align with each other.
